docs/kbs_attestation_protocol.md

```diff
@@ -1,220 +1,418 @@
+of this document scope.
+
+It should be noted that the confidentiality protection provided by HTTPS alone
+is not enough to meet the security requirements of the KBS protocol.
+Therefore, as mentioned above, the key generated by HW-TEE needs to be used to
+encrypt and protect the confidential data.
+
+# `RCAR` semantics
+
+The semantics of attestation defined by KBS is a simple and extensible four-step
+model, which uses JSON structure to organize information. As follows:
+
+1. **Request**: The KBC sends the initial `Request` payload to the KBS, in order
+   to authenticate itself against the KBS, and eventually request resources.
+2. **Challenge**: After receiving the initial `Request` payload, the KBS
+   responds with the `Challenge` payload. This is how the KBS sends the
+   attestation challenge to the KBC. Together with the attestation challenge,
+   the KBS also sends a session identifier to the KBC, as an HTTP Cookie. This
+   session identifier can be used to skip steps 2 and 3 after a successful
+   attestation.
+3. **Attestation**: The KBC replies to the attestation challenge from step 2
+   with an attestation evidence, in order to prove that its environment (HW-TEE)
+   is safe and reliable. The KBC sends an `Attestation` payload to the KBS, that
+   contains the attestation evidence and the HW-TEE generated, ephemeral public
+   key.
+4. **Response**: Upon succesfull attestation, the KBC can request resources by
+   sending HTTP `GET` requests to resource specific endpoints. The KBS responds
+   with a `Response` payload that includes the requested resource data. The
+   resource data is protected by the ephemeral public key passed via the
+   previously sent `Attestation` payload. Within the valid time of the HTTP
+   Cookie generated by the KBS during step 2, the KBC can directly
+   request resources or services from KBS, without going through steps 2 and 3.
+
+# Payloads
+
+## `Request`
 
 The payload format of the request is as follows:
 
-(Note that the `/*...*/` comments are not valid in JSON, and must not be used in real message.)
-
 ```json
 {
-    "request": {
-        /* Attestation protocol version number used by KBC */
-        "version": "0.1.0",
-        /* Type of HW-TEE platforms where KBC is located, such as "tdx", "sev-snp", etc. */
-        "tee": "",
-        /* Access token to avoid multiple attempts triggered by consecutive requests. */
-        "token": "",
-        /* Reserved fields are used to support some special requests sent by HW-TEE. */
-        "extra-params": {}
-    }
+    /* Attestation protocol version number used by KBC */
+    "version": "0.1.0",
+    /*
+     * Type of HW-TEE platforms where KBC is located,
+     * e.g. "intel-tdx", "amd-sev-snp", etc.
+     */
+    "tee": "$tee",
+    /* Reserved fields to support some special requests sent by HW-TEE. */
+    "extra-params": {}
 }
 ```
+<sub>Note that the `/*...*/` comments are not valid in JSON, and must not be
```

Comment:
You could use a ["note" as we do in Kata](https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#notes) to make this stand out (and avoid the need for embedded HTML tags). Same comment for the other instances in this doc.

Reply:
Nice, suggestion applied.